internal/sink/sink.go

```diff
@@ -338,14 +348,28 @@ func (k *Kafka) Update(id, topic string, retries int, servers []string, logger *
 	return nil
 }
 
-func (k Kafkas) Send(secret string, acceptType string, msg *wrp.Message) error {
-	//TODO: discuss with wes and john the default hashing logic
-	//for now: when no hash is given we will just loop through all the kafkas
+func (k KafkaSink) Send(secret string, acceptType string, msg *wrp.Message) error {
 	var errs error
-	for _, kafka := range k {
-		err := kafka.send(secret, acceptType, msg)
-		if err != nil {
-			errs = errors.Join(errs, err)
+	if k.HashField != "" {
+
+		if kafka, ok := k.Kafkas[k.Hash.Get(GetKey(k.HashField, msg))]; ok {
+			err := kafka.send(secret, acceptType, msg)
+			if err != nil {
+				errs = errors.Join(errs, err)
+			}
+
+		} else {
+			errs = fmt.Errorf("could not find kakfa for the related hash %v", k.HashField)
```

Comment:
do we want to send an error here? or instead of the k.HashField if check do we want say if err != nil we go straight to the for loop of allt he kafkas?

Reply:
lets leave this, but we need to think about how to best describe possible error cases like this, i.e. define errors, new metrics, logging (I think this section is good, just add a todo saying we need to flush out the error handling for kafka)

Reply:
technically hashing can't fell as long as there is 1 item so `k.Kafkas[k.Hash.Get(GetKey(k.HashField, msg))]` should always be good

but to make sure of this, we can add a check `len(k.Kafkas) == len(k.Hash)` during the Hash and kafka setup